### PR TITLE
CMake: modernize configuring libvalhalla.pc

### LINKED ISSUE #2903: Install fails to configure libvalhalla.pc: "could not load cache"
When running `cmake --install` it errors out:

```
CMake Error at build/src/cmake_install.cmake:105 (message):
  Configuring libvalhalla.pc failed: Error: could not load cache

Call Stack (most recent call first):
  build/cmake_install.cmake:63 (include)
```

This breaks the build for me.

### PR DESCRIPTION
**Pull request by PureTryOut**
This also fixes the build on at least Alpine Linux.

Resolves #2903

### CODE REVIEW

File: libvalhalla.pc.in [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: libvalhalla
 Description: valhalla c++ library
-Version: @VERSION@
-Libs: -L${libdir} -lvalhalla @deplibs@
+Version: @PROJECT_VERSION@
```
Comment (kevinkreiser) [inline libvalhalla.pc.in:8]: is this actually defined somewhere? the info is located in `valhalla/valhalla.h.in`

Comment (PureTryOut) [reply libvalhalla.pc.in]: Good question, no it wasn't. It was an assumption and a wrong one at that, fixed it. I assumed Valhalla was setting the project version in CMake, but it seems it gets it from CPack and has different variables.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -262,18 +262,11 @@ install(FILES ${valhalla_hdrs}
...
-  set_target_properties(libvalhalla.pc PROPERTIES LINKER_LANGUAGE PkgConfig)
-  set_target_properties(libvalhalla.pc PROPERTIES FOLDER "Library")
-  install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} --build . --target libvalhalla.pc OUTPUT_QUIET ERROR_VARIABLE _err RESULT_VARIABLE _res)
-  if(NOT \${_res} EQUAL 0)
-    message(FATAL_ERROR \"Configuring libvalhalla.pc failed: \${_err}\")
-  endif()")
-  install(FILES ${VALHALLA_BINARY_DIR}/libvalhalla.pc
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
-    COMPONENT development)
+  set(DEPLIBS "<FLAGS> <LINK_FLAGS> <LINK_LIBRARIES>")
```
Comment (PureTryOut) [inline src/CMakeLists.txt:265]: This is going wrong, but I don't fully understand why it worked in the previous method, investigating...

Comment (PureTryOut) [reply src/CMakeLists.txt]: For now I've set it hard-coded to the expected output, not sure if this can be improved somehow...

**Comment by kevinkreiser** [comment]
@PureTryOut ill check this out later on to see if i can figure out why the heck its not working as expected in terms of the dependencies, sorry for the annoyance!

File: libvalhalla.pc.in [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: libvalhalla
 Description: valhalla c++ library
 Version: @VERSION@
-Libs: -L${libdir} -lvalhalla @deplibs@
+Libs: -L${libdir} -lvalhalla -lprotobuf-lite -pthread -lcurl -lz
```
Comment (kevinkreiser) [inline libvalhalla.pc.in:9]: this cant possibly be the full list if `DATA_TOOLS` is enabled (which it most certainly should be in the default case). have you tried the standard print out all the cmake variables trick to see if one of the variables has all the dependencies in it :smile: every time i want to figure out somethign like this i fall back to this: https://stackoverflow.com/questions/9298278/cmake-print-out-all-accessible-variables-in-a-script

Comment (PureTryOut) [reply libvalhalla.pc.in:9]: It isn't no, I currently use valhalla without `DATA_TOOLS` :sweat_smile: 

I have yet to figure out how to do this properly sadly... Will give what you linked a shot

Comment (PureTryOut) [reply libvalhalla.pc.in:9]: Well I printed out everything but still don't really see what I need...
[cmake.log](https://github.com/valhalla/valhalla/files/6128688/cmake.log)

**Pull request closed without merging**